parsl/execution_provider/aws/aws.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+
+import os
+import pprint
+import json
+import time
+
+from parsl.execution_provider.execution_provider_base import ExecutionProvider
+import boto3
+from botocore.exceptions import ClientError
+
+
+WORKER_USERDATA = '''#!/bin/bash
+export JAVA=/usr/local/bin/jdk1.7.0_51/bin
+export SWIFT=/usr/local/bin/swift-trunk/bin
+export PATH=$JAVA:$SWIFT:$PATH
+export WORKER_LOGGING_LEVEL=TRACE
+sudo apt-get install -y python3
+sudo apt-get install -y python3-pip
+sudo pip3 install ipyparallel
+sudo pip3 install parsl
+'''
+
+EC2_LOCATIONS = ['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2']
+
+DEFAULT_LOCATION = 'us-east-2'
+
+
+class EC2(ExecutionProvider):
```

Comment:
Classes, and external facing functions should ideally be commented. Eg. https://github.com/Parsl/parsl/blob/master/parsl/app/app.py